app/models/miq_server/worker_management/monitor.rb

```diff
@@ -45,6 +46,8 @@ def worker_not_responding(w)
   end
 
   def sync_workers
+    cleanup_failed_workers
```

Comment:
Note, I think there's overlap with line 26-28 above and I wonder if the guts of these methods get  moved into `cleanup_failed_workers` 

```
    check_not_responding # already skipped in podified
    check_pending_stop
    clean_worker_records
```

I'm fine with this layout now but I wonder if this needs to be consolidated and duplicate work removed.

What do you think of moving all of this logic to `cleanup_failed_workers` and can we do these things before sync_workers or do they need to happen afterwards?


```ruby
  def cleanup_failed_workers
    if podified?
      # cleanup_failed_pods eventually
    elsif systemd?
      check_not_responding # seems like systemd could eventually handle these in cleanup_failed_systemd_services
      cleanup_failed_systemd_services
    else
      check_not_responding
    end

    check_pending_stop # I'm not sure how this is handled for the spawn/systemd/pods
    clean_worker_records # seems like an artifact we can eliminate if all systems implement a way to get the status of workers outside of a table
  end
```